main/mio.c

```diff
@@ -990,7 +990,10 @@ char *mio_gets (MIO *mio, char *s, size_t size)
 				mio->impl.mem.pos++;
 				if (s[i] == '\n')
 				{
-					i++;
+					if (i > 0 && s[i-1] == '\r')
+						s[i-1] = '\n';
+					else
+						i++;
```

Comment:
The problem is that for CR-only line endings the whole file will be treated as a single line. It's true there shouldn't be much of them these days so maybe it's alright-ish but not correct.